charts/ocis/values.yaml

```diff
@@ -239,134 +541,251 @@ features:
-            # -- Enables the Ingress. Only needed if the office application is not running within the same cluster.
-            enabled: false
-            # -- Domain of the Ingress.
-            domain: ""
-            # -- Ingress class to use.
-            # Uses the default ingress class if not set.
-            ingressClassName:
-            # -- Ingress annotations.
-            annotations: {}
-            # -- Labels for the ingress.
-            labels: {}
-            # -- Ingress TLS configuration.
-            tls: []
-            #  - secretName: chart-example-tls
-            #    hosts:
-            #      - onlyoffice-wopi.owncloud.test
+      # @schema
+      # type: array
+      # required: true
+      # items:
+      #   type: object
+      #   additionalProperties: false
+      #   properties:
+      #     name:
+      #       type: string
+      #       required: true
+      #     product:
+      #       enum: [Collabora, OnlyOffice, Microsoft365, MicrosoftOfficeOnline]
+      #       required: true
+      #     enabled:
+      #       type: boolean
+      #       required: true
+      #     uri:
+      #       type: string
+      #       required: true
+      #     description:
+      #       type: string
+      #       required: false
+      #     iconURI:
+      #       type: string
+      #       required: false
+      #     insecure:
+      #       type: boolean
+      #       required: true
+      #     disableProof:
+      #       type: boolean
+      #       required: true
+      #     secureViewEnabled:
+      #       type: boolean
+      #       required: true
+      #     disableChat:
+      #       type: boolean
+      #       required: true
+      #     ingress:
+      #       type: object
+      #       additionalProperties: false
+      #       properties:
+      #         enabled:
+      #           type: boolean
+      #           required: true
+      #         domain:
+      #           type: string
+      #           required: true
+      #         ingressClassName:
+      #           type: [string, null]
+      #           required: true
+      #         annotations:
+      #           type: object
+      #           required: true
+      #           patternProperties:
+      #            ".*":
+      #              type: string
+      #         labels:
+      #           type: object
+      #           required: true
+      #           patternProperties:
+      #            ".*":
+      #              type: string
+      #         tls:
+      #           type: array
+      #           items:
+      #             type: object
+      #             required: true
+      #             additionalProperties: false
+      #             properties:
+      #               secretName:
+      #                 type: string
+      #                 required: true
+      #               hosts:
+      #                 type: array
+      #                 items:
+      #                   type: string
```

Comment:
We could use https://json-schema.org/understanding-json-schema/reference/string#hostnames here.